cmd/ipfs/main.go

```diff
@@ -82,13 +82,22 @@ func mainRet() int {
 	intrh, ctx := setupInterruptHandler(ctx)
 	defer intrh.Close()
 
-	// Handle `ipfs help'
-	if len(os.Args) == 2 {
-		if os.Args[1] == "help" {
-			os.Args[1] = "-h"
-		} else if os.Args[1] == "--version" {
+	// Handle `ipfs version` or `ipfs help`
+	if len(os.Args) > 1 {
+		// Handle `ipfs --version'
+		if os.Args[1] == "--version" {
 			os.Args[1] = "version"
 		}
+
+		//Handle `ipfs help` and `ipfs help <sub-command>`
+		if os.Args[1] == "help" {
+			if len(os.Args) > 2 {
+				os.Args = append(os.Args[:1], os.Args[2:]...)
+				os.Args = append(os.Args, "-h")
```

Comment:
This should be changed to the long flag `--help` as that applies to `ipfs` itself as well as all subcommands
`-h` only works with `ipfs -h` for instance `ipfs add -h` has a different meaning, but `ipfs add --help` is what we would want.

Reply:
Disregard this, I was confusing `-h` and `-H`.
Although I am wondering which we should use for help, the short description or the long.